src/lib/components/ScrollSentinel.svelte

```diff
@@ -0,0 +1,22 @@
+<script lang="ts">
+  import { layoutContentTopHidden } from "$lib/stores/layout.store";
+  import { onMount } from "svelte";
+
+  // The ScrollSentinel component should be placed right before the content
+  // inside the scrollable container.
+  export let scrollContainer: HTMLElement;
+
+  // To observe when the top leaves the view
+  let element: HTMLElement;
+
+  onMount(() => {
+    const observer = new IntersectionObserver(
+      ([entry]) => layoutContentTopHidden.set(!entry.isIntersecting),
+      { root: scrollContainer, threshold: 0 },
+    );
+    observer.observe(element);
+    return () => observer.disconnect();
+  });
+</script>
+
+<div data-tid="sentinel" bind:this={element}></div>
```

Comment:
What do you think about using the same pseudo hidden style as the placeholder of the InfiniteScroll cmp here (if applicable)?

```
.intersection-observer-target {
    width: 0;
    height: 0;
    opacity: 0;
    visibility: hidden;
  }
```

Reply:
Thank you! That makes sense to avoid accidentally making it visible.

Reply:
Cool. Maybe after this PR we should extract a mixin for it?